### ISSUE TITLE
[Bug] Shims fail if path contains spaces

### ISSUE THREAD

**Issue by jbtls** [ScoopInstaller/Scoop#4728]
## Bug Report

git

### Current Behaviour

Scoop is unable to run command: "Could not create process with command <...>"

### Expected Behaviour

Scoop should update

### Additional context/output

My username has a space in between. 

### Possible Solution

The git shim should be updated to: 
path = "C:\Users\User Name\scoop\apps\git\current\cmd\git.exe"

### System details

**Windows version:** 11

**OS architecture:** 64bit

**PowerShell version:**5.1.22000.282

#### Scoop Configuration

```json
{
    "lastupdate":  "2022-01-27T14:48:14.5181152+01:00",
    "SCOOP_REPO":  "https://github.com/lukesampson/scoop",
    "SCOOP_BRANCH":  "master"
}
```

**Comment by niheaven** [comment]
What if you change scoop to 'develop'?

```console
scoop config SCOOP_BRANCH develop
scoop update
scoop reset git
```

**Comment by jbehmel** [comment]
Same problem. I've just changed the shim myself and it works with the solution given above.

Possible Solution

The git shim should be updated to:
path = "C:\Users\User Name\scoop\apps\git\current\cmd\git.exe"

**Comment by jbehmel** [comment]
Yes, and it works! I've just realized that this problem affects multiple shims (yt-dlp, ffmpeg, ...). It should get fixed there too.

**Comment by rashil2000** [comment]
I'm not able to reproduce this issue. My .shim files have paths with spaces and don't require double quotes.

Can you try these step:
```
scoop config shim kiennq
scoop update
scoop reset git
```

**Comment by niheaven** [comment]
Maybe it's the bug of default shimexe? If so, it should be fixed.

**Comment by rashil2000** [comment]
> Maybe it's the bug of default shimexe? If so, it should be fixed.

We changed the default shim to kiennq some months ago, so that's not likely.

**Comment by jbehmel** [comment]
Hey, sorry for not answering fast.
For me, it seems like it has happened to all shims….

**Comment by JcobCN** [comment]
i got the same problem, that may be related to python env.
I have installed Anaconda, and conda command get the same problem "failed to create process".
Does scoop used python?

**Comment by JcobCN** [comment]
scoop config SCOOP_BRANCH fix-shim
scoop update
scoop reset git
git works well, and need reset all apps.

**Comment by rashil2000** [comment]
It doesn't. 

Does your username contain special/accented characters?

**Comment by JcobCN** [comment]
> 
yes, I modified Scoop installed path at "D:\Program Files\Scoop", so shim path contain spaces character.

**Comment by JcobCN** [comment]
> >
> 
> Yes, I modified Scoop installed path at "D:\Program Files\Scoop", so shim path contain spaces character.
> 
> OS: windows 10

Before today, Scoop works well.

**Comment by rashil2000** [comment]
Here's my environment:

![image](https://user-images.githubusercontent.com/46838874/154818375-3e2581fc-aa80-4405-99d9-8148ff28f350.png)

Can you try using the `develop` branch once?

```
scoop config SCOOP_BRANCH develop
scoop update
scoop reset git
```

**Comment by JcobCN** [comment]
> 



> Here's my environment:
> 
> ![image](https://user-images.githubusercontent.com/46838874/154818375-3e2581fc-aa80-4405-99d9-8148ff28f350.png)
> 
> Can you try using the `develop` branch once?
> 
> ```
> scoop config SCOOP_BRANCH develop
> scoop update
> scoop reset git
> ```

using the `develop`, but does not work.
![image](https://user-images.githubusercontent.com/19268110/154827757-aa0eb71f-54c5-490f-bd6b-0caf0031cb10.png)

**Comment by JcobCN** [comment]
@jbehmel  Have you installed MSVC build tools recently?

**Comment by JcobCN** [comment]
@rashil2000 I think some windows setting changed, My app, windows Terminal, have the same problem, must add `"` in the path.
![image](https://user-images.githubusercontent.com/19268110/154827874-0ab5a8c4-ef92-44f5-9b78-0c2265f284b3.png)

Before install MSVC Buildtools, it works well, `Scoop` works well, and `Anaconda` works well.
MSVC Buildtools change some OS setting?

**Comment by rashil2000** [comment]
I also have MSVC build tools installed, since a long time now.

**Comment by rashil2000** [comment]
For Windows Terminal you will need double quotes because it's json and the binary path has spaces.

**Comment by JcobCN** [comment]
> 



> For Windows Terminal you will need double quotes because it's json and the binary path has spaces.

before that, my  setting is
![image](https://user-images.githubusercontent.com/19268110/154828216-c8915724-0e0e-45b5-8ab5-bc714f1adbb6.png)
and work well.

and now, my setting need add quotes to contain the spaces path.
![image](https://user-images.githubusercontent.com/19268110/154828256-5c50acc6-327a-4703-8c7b-6322e18138c6.png)

**Comment by rashil2000** [comment]
Could you try uninstalling MSVC tools to confirm if that is causing this problem?

**Comment by niheaven** [comment]
If adding quotes doesn't introduce a new bug, #4734 could be merged.

**Comment by rashil2000** [comment]
> Could you try uninstalling MSVC tools to confirm if that is causing this problem?

@JcobCN any updates on this?

**Comment by vieiraa** [comment]
Fixed by https://github.com/71/scoop-better-shimexe/pull/13. 
Will need to add "shim": "71" to your C:\Users\<username>\.config\scoop\config.json.

**Comment by rashil2000** [comment]
Closed in #4734
